src/Shared/CommunicationsUtilities.cs

```diff
@@ -154,6 +160,22 @@ static internal int NodeConnectionTimeout
             get { return GetIntegerVariableOrDefault("MSBUILDNODECONNECTIONTIMEOUT", DefaultNodeConnectionTimeout); }
         }
 
+        private static int? _clrVersion = null;
+        /// <summary>
+        /// Provides cached value of current CLR version
+        /// </summary>
+        private static int ClrVersion
```

Comment:
static variables are given values once per assembly or once per run of the program? I don't think it matters unless you have a very strange program that relies on building different parts from different runtimes.

Reply:
>static class remains in memory for the lifetime of the application domain in which your program resides.
https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/static-classes-and-static-class-members

This is an optimization which was suggested in one of comments. If it causes problems we can revert it.